#!/bin/sh /usr/share/dpatch/dpatch-run

## opt-40_wareagle-icons.dpatch by Torsten Kunkel <vdr@tkunkel.de> (http://vdr.sjur.de)
##
## downloaded from http://vdrportal.de/board/thread.php?postid=266960#post266960
## original filename: wareagle-icon-patch-vdr1.3.20.diff.gz
##
## fixed and extended by Thomas Günther <tom@toms-cafe.de>:
##   - fixed small symbols in fontsml-iso8859-15.c
##   - fixed timers menu: show arrow symbol instead of '!'
##   - added setup option WarEagleIcons to disable the icons (e.g. for skins
##     with truetype fonts) - as default the icons are enabled
##   http://toms-cafe.de/vdr/download/vdr-wareagle-icons+setup-1.3.22.diff
##
## Thomas Günther <tom@toms-cafe.de>:
##   - adapted to sort-options patch
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Adds some icons to timer, recording and channel list.

@DPATCH@
diff -Naur vdr-1.3.22/config.c vdr-1.3.22-icon/config.c
--- vdr-1.3.22/config.c	2005-02-20 13:52:59.000000000 +0100
+++ vdr-1.3.22-icon/config.c	2005-03-06 01:54:41.000000000 +0100
@@ -247,6 +247,7 @@
   OSDLanguage = 0;
   strcpy(OSDSkin, "sttng");
   strcpy(OSDTheme, "default");
+  WarEagleIcons = 1;
   PrimaryDVB = 1;
   ShowInfoOnChSwitch = 1;
   MenuScrollPage = 1;
@@ -404,6 +405,7 @@
   if      (!strcasecmp(Name, "OSDLanguage"))         OSDLanguage        = atoi(Value);
   else if (!strcasecmp(Name, "OSDSkin"))             strn0cpy(OSDSkin, Value, MaxSkinName);
   else if (!strcasecmp(Name, "OSDTheme"))            strn0cpy(OSDTheme, Value, MaxThemeName);
+  else if (!strcasecmp(Name, "WarEagleIcons"))       WarEagleIcons      = atoi(Value);
   else if (!strcasecmp(Name, "PrimaryDVB"))          PrimaryDVB         = atoi(Value);
   else if (!strcasecmp(Name, "ShowInfoOnChSwitch"))  ShowInfoOnChSwitch = atoi(Value);
   else if (!strcasecmp(Name, "MenuScrollPage"))      MenuScrollPage     = atoi(Value);
@@ -468,6 +470,7 @@
   Store("OSDLanguage",        OSDLanguage);
   Store("OSDSkin",            OSDSkin);
   Store("OSDTheme",           OSDTheme);
+  Store("WarEagleIcons",      WarEagleIcons);
   Store("PrimaryDVB",         PrimaryDVB);
   Store("ShowInfoOnChSwitch", ShowInfoOnChSwitch);
   Store("MenuScrollPage",     MenuScrollPage);
diff -Naur vdr-1.3.22/config.h vdr-1.3.22-icon/config.h
--- vdr-1.3.22/config.h	2005-02-20 13:50:37.000000000 +0100
+++ vdr-1.3.22-icon/config.h	2005-03-06 01:52:21.000000000 +0100
@@ -208,6 +208,7 @@
   int OSDLanguage;
   char OSDSkin[MaxSkinName];
   char OSDTheme[MaxThemeName];
+  int WarEagleIcons;
   int PrimaryDVB;
   int ShowInfoOnChSwitch;
   int MenuScrollPage;
diff -Naur vdr-1.3.22/fontosd-iso8859-15.c vdr-1.3.22-icon/fontosd-iso8859-15.c
--- vdr-1.3.22/fontosd-iso8859-15.c	2004-12-18 17:14:49.000000000 +0100
+++ vdr-1.3.22-icon/fontosd-iso8859-15.c	2005-03-04 19:58:28.000000000 +0100
@@ -6210,35 +6210,65 @@
      0x00000000,  // ......
   },
   {             // 239
-     6, 27,
-     0x00000000,  // ......
-     0x00000000,  // ......
-     0x00000000,  // ......
-     0x00000000,  // ......
-     0x00000000,  // ......
-     0x00000000,  // ......
-     0x00000033,  // **..**
-     0x00000033,  // **..**
-     0x00000000,  // ......
-     0x00000000,  // ......
-     0x0000000C,  // ..**..
-     0x0000000C,  // ..**..
-     0x0000000C,  // ..**..
-     0x0000000C,  // ..**..
-     0x0000000C,  // ..**..
-     0x0000000C,  // ..**..
-     0x0000000C,  // ..**..
-     0x0000000C,  // ..**..
-     0x0000000C,  // ..**..
-     0x0000000C,  // ..**..
-     0x0000000C,  // ..**..
-     0x0000000C,  // ..**..
-     0x00000000,  // ......
-     0x00000000,  // ......
-     0x00000000,  // ......
-     0x00000000,  // ......
-     0x00000000,  // ......
-  },
+       16, 27,
+       0x0,    // ................................
+       0x0,    // ................................
+       0x0,    // ................................
+       0x0,    // ................................
+       0x6006, // .................##..........##.
+       0x700e, // .................###........###.
+       0x381c, // ..................###......###..
+       0x1c38, // ...................###....###...
+       0xe70,  // ....................###..###....
+       0x7e0,  // .....................######.....
+       0x3ffc, // ..................############..
+       0x7ffe, // .................##############.
+       0x700e, // .................###........###.
+       0x6006, // .................##..........##.
+       0x6c06, // .................##.##.......##.
+       0x6ff6, // .................##.########.##.
+       0x6ff6, // .................##.########.##.
+       0x6db6, // .................##.##.##.##.##.
+       0x61b6, // .................##....##.##.##.
+       0x61b6, // .................##....##.##.##.
+       0x6006, // .................##..........##.
+       0x700e, // .................###........###.
+       0x7ffe, // .................##############.
+       0x3ffc, // ..................############..
+       0x0,    // ................................
+       0x0,    // ................................
+       0x0,    // ................................
+  },
+//  {             // 239
+//     6, 27,
+//     0x00000000,  // ......
+//     0x00000000,  // ......
+//     0x00000000,  // ......
+//     0x00000000,  // ......
+//     0x00000000,  // ......
+//     0x00000000,  // ......
+//     0x00000033,  // **..**
+//     0x00000033,  // **..**
+//     0x00000000,  // ......
+//     0x00000000,  // ......
+//     0x0000000C,  // ..**..
+//     0x0000000C,  // ..**..
+//     0x0000000C,  // ..**..
+//     0x0000000C,  // ..**..
+//     0x0000000C,  // ..**..
+//     0x0000000C,  // ..**..
+//     0x0000000C,  // ..**..
+//     0x0000000C,  // ..**..
+//     0x0000000C,  // ..**..
+//     0x0000000C,  // ..**..
+//     0x0000000C,  // ..**..
+//     0x0000000C,  // ..**..
+//     0x00000000,  // ......
+//     0x00000000,  // ......
+//     0x00000000,  // ......
+//     0x00000000,  // ......
+//     0x00000000,  // ......
+//  },
   {             // 240
      13, 27,
      0x00000000,  // .............
@@ -6360,65 +6390,125 @@
      0x00000000,  // .............
   },
   {             // 244
-     13, 27,
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x000000E0,  // .....***.....
-     0x000001A0,  // ....**.*.....
-     0x00000130,  // ....*..**....
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x000001F0,  // ....*****....
-     0x000007F8,  // ..********...
-     0x0000061C,  // ..**....***..
-     0x00000C0E,  // .**......***.
-     0x00000C06,  // .**.......**.
-     0x00000C06,  // .**.......**.
-     0x00000C06,  // .**.......**.
-     0x00000C06,  // .**.......**.
-     0x00000C0E,  // .**......***.
-     0x0000061C,  // ..**....***..
-     0x000007F8,  // ..********...
-     0x000001F0,  // ....*****....
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-  },
+    16, 27,
+    0x0,       // ................................
+    0x0,       // ................................
+    0x0,       // ................................
+    0x0,       // ................................
+    0x0,       // ................................
+    0x1c,      // ...........................###..
+    0x38,      // ..........................###...
+    0x70,      // .........................###....
+    0xe0,      // ........................###.....
+    0x1c0,     // .......................###......
+    0x380,     // ......................###.......
+    0x300,     // ......................##........
+    0x300,     // ......................##........
+    0x7ffe,    // .................##############.
+    0x7ffe,    // .................##############.
+    0x6006,    // .................##..........##.
+    0x6006,    // .................##..........##.
+    0x6006,    // .................##..........##.
+    0x6006,    // .................##..........##.
+    0x6006,    // .................##..........##.
+    0x6006,    // .................##..........##.
+    0x6006,    // .................##..........##.
+    0x7ffe,    // .................##############.
+    0x7ffe,    // .................##############.
+    0x0,       // ................................
+    0x0,       // ................................
+    0x0,       // ................................
+  },
+//  {             // 244
+//     13, 27,
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x000000E0,  // .....***.....
+//     0x000001A0,  // ....**.*.....
+//     0x00000130,  // ....*..**....
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x000001F0,  // ....*****....
+//     0x000007F8,  // ..********...
+//     0x0000061C,  // ..**....***..
+//     0x00000C0E,  // .**......***.
+//     0x00000C06,  // .**.......**.
+//     0x00000C06,  // .**.......**.
+//     0x00000C06,  // .**.......**.
+//     0x00000C06,  // .**.......**.
+//     0x00000C0E,  // .**......***.
+//     0x0000061C,  // ..**....***..
+//     0x000007F8,  // ..********...
+//     0x000001F0,  // ....*****....
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//  },
   {             // 245
-     13, 27,
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000180,  // ....**.......
-     0x000003F8,  // ...*******...
-     0x00000270,  // ...*..***....
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x000001F0,  // ....*****....
-     0x000007F8,  // ..********...
-     0x0000061C,  // ..**....***..
-     0x00000C0E,  // .**......***.
-     0x00000C06,  // .**.......**.
-     0x00000C06,  // .**.......**.
-     0x00000C06,  // .**.......**.
-     0x00000C06,  // .**.......**.
-     0x00000C0E,  // .**......***.
-     0x0000061C,  // ..**....***..
-     0x000007F8,  // ..********...
-     0x000001F0,  // ....*****....
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-  },
+    16, 27,
+    0x0,       // ................................
+    0x0,       // ................................
+    0x0,       // ................................
+    0x0,       // ................................
+    0x6006,    // .................##..........##.
+    0x700e,    // .................###........###.
+    0x381c,    // ..................###......###..
+    0x1c38,    // ...................###....###...
+    0xe70,     // ....................###..###....
+    0x7e0,     // .....................######.....
+    0x3ffc,    // ..................############..
+    0x7ffe,    // .................##############.
+    0x700e,    // .................###........###.
+    0x6006,    // .................##..........##.
+    0x6006,    // .................##..........##.
+    0x6006,    // .................##..........##.
+    0x6006,    // .................##..........##.
+    0x6006,    // .................##..........##.
+    0x6006,    // .................##..........##.
+    0x6006,    // .................##..........##.
+    0x6006,    // .................##..........##.
+    0x700e,    // .................###........###.
+    0x7ffe,    // .................##############.
+    0x3ffc,    // ..................############..
+    0x0,       // ................................
+    0x0,       // ................................
+    0x0,       // ................................
+  },
+//  {             // 245
+//     13, 27,
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000180,  // ....**.......
+//     0x000003F8,  // ...*******...
+//     0x00000270,  // ...*..***....
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x000001F0,  // ....*****....
+//     0x000007F8,  // ..********...
+//     0x0000061C,  // ..**....***..
+//     0x00000C0E,  // .**......***.
+//     0x00000C06,  // .**.......**.
+//     0x00000C06,  // .**.......**.
+//     0x00000C06,  // .**.......**.
+//     0x00000C06,  // .**.......**.
+//     0x00000C0E,  // .**......***.
+//     0x0000061C,  // ..**....***..
+//     0x000007F8,  // ..********...
+//     0x000001F0,  // ....*****....
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//  },
   {             // 246
      13, 27,
      0x00000000,  // .............
@@ -6510,95 +6600,185 @@
      0x00000000,  // ..............
   },
   {             // 249
-     13, 27,
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000380,  // ...***.......
-     0x000000C0,  // .....**......
-     0x00000040,  // ......*......
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000C0C,  // .**......**..
-     0x00000C0C,  // .**......**..
-     0x00000C0C,  // .**......**..
-     0x00000C0C,  // .**......**..
-     0x00000C0C,  // .**......**..
-     0x00000C0C,  // .**......**..
-     0x00000C0C,  // .**......**..
-     0x00000C0C,  // .**......**..
-     0x00000C0C,  // .**......**..
-     0x00000C1C,  // .**.....***..
-     0x00000FFC,  // .**********..
-     0x000003E0,  // ...*****.....
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-  },
+     16, 27,
+       0x0,    // ................
+       0xf800, // #####...........
+       0xfc00, // ######..........
+       0xcc00, // ##..##..........
+       0xfc00, // ######..........
+       0xf000, // ####............
+       0xf800, // #####...........
+       0xfc00, // ######..........
+       0xdc00, // ##.###..........
+       0x0,    // ................
+       0x1f80, // ...######.......
+       0x1f80, // ...######.......
+       0x1800, // ...##...........
+       0x1e00, // ...####.........
+       0x1e00, // ...####.........
+       0x1800, // ...##...........
+       0x1f80, // ...######.......
+       0x1f80, // ...######.......
+       0x0,    // ................
+       0x3c0,  // ......####......
+       0x7e0,  // .....######.....
+       0xe60,  // ....###..##.....
+       0xc00,  // ....##..........
+       0xc00,  // ....##..........
+       0xe60,  // ....###..##.....
+       0x7e0,  // .....######.....
+       0x3c0,  // ......####......
+  },
+//  {             // 249
+//     13, 27,
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000380,  // ...***.......
+//     0x000000C0,  // .....**......
+//     0x00000040,  // ......*......
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000C0C,  // .**......**..
+//     0x00000C0C,  // .**......**..
+//     0x00000C0C,  // .**......**..
+//     0x00000C0C,  // .**......**..
+//     0x00000C0C,  // .**......**..
+//     0x00000C0C,  // .**......**..
+//     0x00000C0C,  // .**......**..
+//     0x00000C0C,  // .**......**..
+//     0x00000C0C,  // .**......**..
+//     0x00000C1C,  // .**.....***..
+//     0x00000FFC,  // .**********..
+//     0x000003E0,  // ...*****.....
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//  },
   {             // 250
-     13, 27,
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000070,  // ......***....
-     0x00000060,  // ......**.....
-     0x000000C0,  // .....**......
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000C0C,  // .**......**..
-     0x00000C0C,  // .**......**..
-     0x00000C0C,  // .**......**..
-     0x00000C0C,  // .**......**..
-     0x00000C0C,  // .**......**..
-     0x00000C0C,  // .**......**..
-     0x00000C0C,  // .**......**..
-     0x00000C0C,  // .**......**..
-     0x00000C0C,  // .**......**..
-     0x00000C1C,  // .**.....***..
-     0x00000FFC,  // .**********..
-     0x000003E0,  // ...*****.....
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-  },
+       16, 27,
+       0x0,    // ................
+       0xc600, // ##...##.........
+       0xe600, // ###..##.........
+       0xf600, // ####.##.........
+       0xf600, // ####.##.........
+       0xde00, // ##.####.........
+       0xde00, // ##.####.........
+       0xce00, // ##..###.........
+       0xc600, // ##...##.........
+       0x0,    // ................
+       0x1fc0, // ...#######......
+       0x1fc0, // ...#######......
+       0x1800, // ...##...........
+       0x1f00, // ...#####........
+       0x1f00, // ...#####........
+       0x1800, // ...##...........
+       0x1fc0, // ...#######......
+       0x1fc0, // ...#######......
+       0x0,    // ................
+       0x318,  // ......##...##...
+       0x318,  // ......##...##...
+       0x318,  // ......##...##...
+       0x318,  // ......##...##...
+       0x318,  // ......##...##...
+       0x318,  // ......##...##...
+       0x3f8,  // ......#######...
+       0x1f0,  // .......#####....
+  },
+//  {             // 250
+//     13, 27,
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000070,  // ......***....
+//     0x00000060,  // ......**.....
+//     0x000000C0,  // .....**......
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000C0C,  // .**......**..
+//     0x00000C0C,  // .**......**..
+//     0x00000C0C,  // .**......**..
+//     0x00000C0C,  // .**......**..
+//     0x00000C0C,  // .**......**..
+//     0x00000C0C,  // .**......**..
+//     0x00000C0C,  // .**......**..
+//     0x00000C0C,  // .**......**..
+//     0x00000C0C,  // .**......**..
+//     0x00000C1C,  // .**.....***..
+//     0x00000FFC,  // .**********..
+//     0x000003E0,  // ...*****.....
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//  },
   {             // 251
-     13, 27,
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x000000E0,  // .....***.....
-     0x000001A0,  // ....**.*.....
-     0x00000130,  // ....*..**....
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000C0C,  // .**......**..
-     0x00000C0C,  // .**......**..
-     0x00000C0C,  // .**......**..
-     0x00000C0C,  // .**......**..
-     0x00000C0C,  // .**......**..
-     0x00000C0C,  // .**......**..
-     0x00000C0C,  // .**......**..
-     0x00000C0C,  // .**......**..
-     0x00000C0C,  // .**......**..
-     0x00000C1C,  // .**.....***..
-     0x00000FFC,  // .**********..
-     0x000003E0,  // ...*****.....
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-  },
+       16, 27,
+       0x0,    // ................
+       0x0,    // ................
+       0x78,   // ..........####..
+       0xfc,   // .........######.
+       0x1e6,  // ........####..##
+       0x1fe,  // ........########
+       0x1de,  // ........###.####
+       0x1e0,  // ........####....
+       0x70f8, // ..###....#####..
+       0xf8c0, // .#####...##.....
+       0x19dc0,        // ##..###.###.....
+       0x18f80,        // ##...#####......
+       0x1c780,        // ###...####......
+       0xcf00, // .##..####.......
+       0x1f98, // ....######..##..
+       0x39f0, // ...###..#####...
+       0x70e0, // ..###....###....
+       0xe000, // .###............
+       0x1e000,        // ####............
+       0x1b000,        // ##.##...........
+       0x19800,        // ##..##..........
+       0xcc00, // .##..##.........
+       0x6600, // ..##..##........
+       0x3b80, // ...###.###......
+       0x3b80, // ...###.###......
+       0x0,    // ................
+       0x0,    // ................
+  },
+//  {             // 251
+//     13, 27,
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x000000E0,  // .....***.....
+//     0x000001A0,  // ....**.*.....
+//     0x00000130,  // ....*..**....
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000C0C,  // .**......**..
+//     0x00000C0C,  // .**......**..
+//     0x00000C0C,  // .**......**..
+//     0x00000C0C,  // .**......**..
+//     0x00000C0C,  // .**......**..
+//     0x00000C0C,  // .**......**..
+//     0x00000C0C,  // .**......**..
+//     0x00000C0C,  // .**......**..
+//     0x00000C0C,  // .**......**..
+//     0x00000C1C,  // .**.....***..
+//     0x00000FFC,  // .**********..
+//     0x000003E0,  // ...*****.....
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//  },
   {             // 252
      13, 27,
      0x00000000,  // .............
@@ -6630,93 +6810,196 @@
      0x00000000,  // .............
   },
   {             // 253
-     12, 27,
-     0x00000000,  // ............
-     0x00000000,  // ............
-     0x00000000,  // ............
-     0x00000000,  // ............
-     0x00000000,  // ............
-     0x00000038,  // ......***...
-     0x00000070,  // .....***....
-     0x00000040,  // .....*......
-     0x00000000,  // ............
-     0x00000000,  // ............
-     0x00000606,  // .**......**.
-     0x0000060E,  // .**.....***.
-     0x0000060C,  // .**.....**..
-     0x0000030C,  // ..**....**..
-     0x00000318,  // ..**...**...
-     0x00000318,  // ..**...**...
-     0x00000198,  // ...**..**...
-     0x000001B0,  // ...**.**....
-     0x000001B0,  // ...**.**....
-     0x000000F0,  // ....****....
-     0x000000E0,  // ....***.....
-     0x000000E0,  // ....***.....
-     0x000000C0,  // ....**......
-     0x000000C0,  // ....**......
-     0x000001C0,  // ...***......
-     0x00000780,  // .****.......
-     0x00000700,  // .***........
-  },
+        16, 27,
+        0x0,    // ................
+        0x0,    // ................
+        0x0,    // ................
+        0x0,    // ................
+        0x0,    // ................
+        0x6006, // .##..........##.
+        0xf3cf, // ####..####..####
+        0xf7ef, // ####.######.####
+        0x3c3c, // ..####....####..
+        0x1998, // ...##..##..##...
+        0x318c, // ..##...##...##..
+        0x318c, // ..##...##...##..
+        0x30cc, // ..##....##..##..
+        0x38dc, // ..###...##.###..
+        0x1c38, // ...###....###...
+        0xff0,  // ....########....
+        0xff0,  // ....########....
+        0xc30,  // ....##....##....
+        0x1818, // ...##......##...
+        0x381c, // ..###......###..
+        0xfc3f, // ######....######
+        0xfc3f, // ######....######
+        0x0,    // ................
+        0x0,    // ................
+        0x0,    // ................
+        0x0,    // ................
+        0x0,    // ................
+  },
+//  {             // 253
+//     12, 27,
+//     0x00000000,  // ............
+//     0x00000000,  // ............
+//     0x00000000,  // ............
+//     0x00000000,  // ............
+//     0x00000000,  // ............
+//     0x00000038,  // ......***...
+//     0x00000070,  // .....***....
+//     0x00000040,  // .....*......
+//     0x00000000,  // ............
+//     0x00000000,  // ............
+//     0x00000606,  // .**......**.
+//     0x0000060E,  // .**.....***.
+//     0x0000060C,  // .**.....**..
+//     0x0000030C,  // ..**....**..
+//     0x00000318,  // ..**...**...
+//     0x00000318,  // ..**...**...
+//     0x00000198,  // ...**..**...
+//     0x000001B0,  // ...**.**....
+//     0x000001B0,  // ...**.**....
+//     0x000000F0,  // ....****....
+//     0x000000E0,  // ....***.....
+//     0x000000E0,  // ....***.....
+//     0x000000C0,  // ....**......
+//     0x000000C0,  // ....**......
+//     0x000001C0,  // ...***......
+//     0x00000780,  // .****.......
+//     0x00000700,  // .***........
+//  },
   {             // 254
-     13, 27,
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000000,  // .............
-     0x00000C00,  // .**..........
-     0x00000C00,  // .**..........
-     0x00000C00,  // .**..........
-     0x00000C00,  // .**..........
-     0x00000DF0,  // .**.*****....
-     0x00000FFC,  // .**********..
-     0x00000F1C,  // .****...***..
-     0x00000E0E,  // .***.....***.
-     0x00000C06,  // .**.......**.
-     0x00000C06,  // .**.......**.
-     0x00000C06,  // .**.......**.
-     0x00000C06,  // .**.......**.
-     0x00000E0E,  // .***.....***.
-     0x00000F1C,  // .****...***..
-     0x00000FFC,  // .**********..
-     0x00000CF0,  // .**..****....
-     0x00000C00,  // .**..........
-     0x00000C00,  // .**..........
-     0x00000C00,  // .**..........
-     0x00000C00,  // .**..........
-     0x00000C00,  // .**..........
-  },
+        16, 27,
+        0x0,    // ................
+        0x0,    // ................
+        0x0,    // ................
+        0x0,    // ................
+        0x0,    // ................
+        0x4,    // ..............#.
+        0x1180c,        // #...##.......##.
+        0x19c18,        // ##..###.....##..
+        0xd630, // .##.#.##...##...
+        0x7360, // ..###..##.##....
+        0x31c0, // ...##...###.....
+        0x19f0, // ....##..#####...
+        0x1f50, // ....#####.#.#...
+        0x1650, // ....#.##..#.#...
+        0x1f50, // ....#####.#.#...
+        0x19f0, // ....##..#####...
+        0x31c0, // ...##...###.....
+        0x7360, // ..###..##.##....
+        0xd630, // .##.#.##...##...
+        0x19c18,        // ##..###.....##..
+        0x1180c,        // #...##.......##.
+        0x4,    // ..............#.
+        0x0,    // ................
+        0x0,    // ................
+        0x0,    // ................
+        0x0,    // ................
+        0x0,    // ................
+  },
+//  {             // 254
+//     13, 27,
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000000,  // .............
+//     0x00000C00,  // .**..........
+//     0x00000C00,  // .**..........
+//     0x00000C00,  // .**..........
+//     0x00000C00,  // .**..........
+//     0x00000DF0,  // .**.*****....
+//     0x00000FFC,  // .**********..
+//     0x00000F1C,  // .****...***..
+//     0x00000E0E,  // .***.....***.
+//     0x00000C06,  // .**.......**.
+//     0x00000C06,  // .**.......**.
+//     0x00000C06,  // .**.......**.
+//     0x00000C06,  // .**.......**.
+//     0x00000E0E,  // .***.....***.
+//     0x00000F1C,  // .****...***..
+//     0x00000FFC,  // .**********..
+//     0x00000CF0,  // .**..****....
+//     0x00000C00,  // .**..........
+//     0x00000C00,  // .**..........
+//     0x00000C00,  // .**..........
+//     0x00000C00,  // .**..........
+//     0x00000C00,  // .**..........
+//  },
   {             // 255
-     12, 27,
-     0x00000000,  // ............
-     0x00000000,  // ............
-     0x00000000,  // ............
-     0x00000000,  // ............
-     0x00000000,  // ............
-     0x00000000,  // ............
-     0x000001B8,  // ...**.***...
-     0x000001B8,  // ...**.***...
-     0x00000000,  // ............
-     0x00000000,  // ............
-     0x00000606,  // .**......**.
-     0x0000060E,  // .**.....***.
-     0x0000060C,  // .**.....**..
-     0x0000030C,  // ..**....**..
-     0x00000318,  // ..**...**...
-     0x00000318,  // ..**...**...
-     0x00000198,  // ...**..**...
-     0x000001B0,  // ...**.**....
-     0x000001B0,  // ...**.**....
-     0x000000F0,  // ....****....
-     0x000000E0,  // ....***.....
-     0x000000E0,  // ....***.....
-     0x000000C0,  // ....**......
-     0x000000C0,  // ....**......
-     0x000001C0,  // ...***......
-     0x00000780,  // .****.......
-     0x00000700,  // .***........
-  },
+     16, 27,
+        0x0,    // ................
+        0x0,    // ................
+        0x7f80, // ..########......
+        0xffc0, // .##########.....
+        0x1ffe0,        // ############....
+        0x1f3e0,        // #####..#####....
+        0x1e1e0,        // ####....####....
+        0x1c0e0,        // ###......###....
+        0x1c0e0,        // ###......###....
+        0x1c0e0,        // ###......###....
+        0x1c0e0,        // ###......###....
+        0x1c0e0,        // ###......###....
+        0x1c0e0,        // ###......###....
+        0x1c0e0,        // ###......###....
+        0x1c0e0,        // ###......###....
+        0x1c0e0,        // ###......###....
+        0x1c7fc,        // ###...#########.
+        0x1c3f8,        // ###....#######..
+        0x1c1f0,        // ###.....#####...
+        0x1c0e0,        // ###......###....
+        0x40,   // ..........#.....
+        0x0,    // ................
+        0x0,    // ................
+        0x0,    // ................
+        0x0,    // ................
+        0x0,    // ................
+        0x0,    // ................
+  },
+//  {             // 255
+//     12, 27,
+//     0x00000000,  // ............
+//     0x00000000,  // ............
+//     0x00000000,  // ............
+//     0x00000000,  // ............
+//     0x00000000,  // ............
+//     0x00000000,  // ............
+//     0x000001B8,  // ...**.***...
+//     0x000001B8,  // ...**.***...
+//     0x00000000,  // ............
+//     0x00000000,  // ............
+//     0x00000606,  // .**......**.
+//     0x0000060E,  // .**.....***.
+//     0x0000060C,  // .**.....**..
+//     0x0000030C,  // ..**....**..
+//     0x00000318,  // ..**...**...
+//     0x00000318,  // ..**...**...
+//     0x00000198,  // ...**..**...
+//     0x000001B0,  // ...**.**....
+//     0x000001B0,  // ...**.**....
+//     0x000000F0,  // ....****....
+//     0x000000E0,  // ....***.....
+//     0x000000E0,  // ....***.....
+//     0x000000C0,  // ....**......
+//     0x000000C0,  // ....**......
+//     0x000001C0,  // ...***......
+//     0x00000780,  // .****.......
+//     0x00000700,  // .***........
+//  },
   };
+
+// WarEagle-Symbol-Patch
+// Zeichen     Symbol
+//
+//     255     Pfeil
+//     254     Lautsprecher
+//     253     Wecker
+//     251     Maennchen
+//     250     "NEU"
+//     249     "REC"
+//     244     Weiter-Links
+//     245     TV
+//     239     TV-verschluesselt
diff -Naur vdr-1.3.22/fontsml-iso8859-15.c vdr-1.3.22-icon/fontsml-iso8859-15.c
--- vdr-1.3.22/fontsml-iso8859-15.c	2004-12-18 17:14:51.000000000 +0100
+++ vdr-1.3.22-icon/fontsml-iso8859-15.c	2005-03-07 03:48:59.000000000 +0100
@@ -5175,30 +5175,55 @@
      0x00000000,  // ......
   },
   {             // 239
-     5, 22,
-     0x00000000,  // .....
-     0x00000000,  // .....
-     0x00000000,  // .....
-     0x00000000,  // .....
-     0x0000001B,  // **.**
-     0x0000001B,  // **.**
-     0x00000000,  // .....
-     0x00000000,  // .....
-     0x0000000C,  // .**..
-     0x0000000C,  // .**..
-     0x0000000C,  // .**..
-     0x0000000C,  // .**..
-     0x0000000C,  // .**..
-     0x0000000C,  // .**..
-     0x0000000C,  // .**..
-     0x0000000C,  // .**..
-     0x0000000C,  // .**..
-     0x0000000C,  // .**..
-     0x00000000,  // .....
-     0x00000000,  // .....
-     0x00000000,  // .....
-     0x00000000,  // .....
-  },
+       16, 22,
+       0x0,    // ................................
+       0x6006, // .................##..........##.
+       0x700e, // .................###........###.
+       0x381c, // ..................###......###..
+       0x1c38, // ...................###....###...
+       0xe70,  // ....................###..###....
+       0x7e0,  // .....................######.....
+       0x3ffc, // ..................############..
+       0x7ffe, // .................##############.
+       0x700e, // .................###........###.
+       0x6006, // .................##..........##.
+       0x6c06, // .................##.##.......##.
+       0x6ff6, // .................##.########.##.
+       0x6ff6, // .................##.########.##.
+       0x6db6, // .................##.##.##.##.##.
+       0x61b6, // .................##....##.##.##.
+       0x61b6, // .................##....##.##.##.
+       0x6006, // .................##..........##.
+       0x700e, // .................###........###.
+       0x7ffe, // .................##############.
+       0x3ffc, // ..................############..
+       0x0,    // ................................
+  },
+//  {             // 239
+//     5, 22,
+//     0x00000000,  // .....
+//     0x00000000,  // .....
+//     0x00000000,  // .....
+//     0x00000000,  // .....
+//     0x0000001B,  // **.**
+//     0x0000001B,  // **.**
+//     0x00000000,  // .....
+//     0x00000000,  // .....
+//     0x0000000C,  // .**..
+//     0x0000000C,  // .**..
+//     0x0000000C,  // .**..
+//     0x0000000C,  // .**..
+//     0x0000000C,  // .**..
+//     0x0000000C,  // .**..
+//     0x0000000C,  // .**..
+//     0x0000000C,  // .**..
+//     0x0000000C,  // .**..
+//     0x0000000C,  // .**..
+//     0x00000000,  // .....
+//     0x00000000,  // .....
+//     0x00000000,  // .....
+//     0x00000000,  // .....
+//  },
   {             // 240
      11, 22,
      0x00000000,  // ...........
@@ -5300,55 +5325,105 @@
      0x00000000,  // ...........
   },
   {             // 244
-     11, 22,
-     0x00000000,  // ...........
-     0x00000000,  // ...........
-     0x00000000,  // ...........
-     0x00000000,  // ...........
-     0x00000060,  // ....**.....
-     0x000000F0,  // ...****....
-     0x00000198,  // ..**..**...
-     0x00000000,  // ...........
-     0x000000F8,  // ...*****...
-     0x000001FC,  // ..*******..
-     0x0000018C,  // ..**...**..
-     0x00000306,  // .**.....**.
-     0x00000306,  // .**.....**.
-     0x00000306,  // .**.....**.
-     0x00000306,  // .**.....**.
-     0x0000018C,  // ..**...**..
-     0x000001FC,  // ..*******..
-     0x000000F8,  // ...*****...
-     0x00000000,  // ...........
-     0x00000000,  // ...........
-     0x00000000,  // ...........
-     0x00000000,  // ...........
-  },
+    16, 22,
+    0x0,        // ................................
+    0x0,        // ................................
+    0x1c,       // ...........................###..
+    0x38,       // ..........................###...
+    0x70,       // .........................###....
+    0xe0,       // ........................###.....
+    0x1c0,      // .......................###......
+    0x380,      // ......................###.......
+    0x300,      // ......................##........
+    0x300,      // ......................##........
+    0x7ffe,     // .................##############.
+    0x7ffe,     // .................##############.
+    0x6006,     // .................##..........##.
+    0x6006,     // .................##..........##.
+    0x6006,     // .................##..........##.
+    0x6006,     // .................##..........##.
+    0x6006,     // .................##..........##.
+    0x6006,     // .................##..........##.
+    0x6006,     // .................##..........##.
+    0x7ffe,     // .................##############.
+    0x7ffe,     // .................##############.
+    0x0,        // ................................
+  },
+//  {             // 244
+//     11, 22,
+//     0x00000000,  // ...........
+//     0x00000000,  // ...........
+//     0x00000000,  // ...........
+//     0x00000000,  // ...........
+//     0x00000060,  // ....**.....
+//     0x000000F0,  // ...****....
+//     0x00000198,  // ..**..**...
+//     0x00000000,  // ...........
+//     0x000000F8,  // ...*****...
+//     0x000001FC,  // ..*******..
+//     0x0000018C,  // ..**...**..
+//     0x00000306,  // .**.....**.
+//     0x00000306,  // .**.....**.
+//     0x00000306,  // .**.....**.
+//     0x00000306,  // .**.....**.
+//     0x0000018C,  // ..**...**..
+//     0x000001FC,  // ..*******..
+//     0x000000F8,  // ...*****...
+//     0x00000000,  // ...........
+//     0x00000000,  // ...........
+//     0x00000000,  // ...........
+//     0x00000000,  // ...........
+//  },
   {             // 245
-     11, 22,
-     0x00000000,  // ...........
-     0x00000000,  // ...........
-     0x00000000,  // ...........
-     0x00000000,  // ...........
-     0x000000C8,  // ...**..*...
-     0x00000168,  // ..*.**.*...
-     0x00000130,  // ..*..**....
-     0x00000000,  // ...........
-     0x000000F8,  // ...*****...
-     0x000001FC,  // ..*******..
-     0x0000018C,  // ..**...**..
-     0x00000306,  // .**.....**.
-     0x00000306,  // .**.....**.
-     0x00000306,  // .**.....**.
-     0x00000306,  // .**.....**.
-     0x0000018C,  // ..**...**..
-     0x000001FC,  // ..*******..
-     0x000000F8,  // ...*****...
-     0x00000000,  // ...........
-     0x00000000,  // ...........
-     0x00000000,  // ...........
-     0x00000000,  // ...........
-  },
+    16, 22,
+    0x0,        // ................................
+    0x6006,     // .................##..........##.
+    0x700e,     // .................###........###.
+    0x381c,     // ..................###......###..
+    0x1c38,     // ...................###....###...
+    0xe70,      // ....................###..###....
+    0x7e0,      // .....................######.....
+    0x3ffc,     // ..................############..
+    0x7ffe,     // .................##############.
+    0x700e,     // .................###........###.
+    0x6006,     // .................##..........##.
+    0x6006,     // .................##..........##.
+    0x6006,     // .................##..........##.
+    0x6006,     // .................##..........##.
+    0x6006,     // .................##..........##.
+    0x6006,     // .................##..........##.
+    0x6006,     // .................##..........##.
+    0x6006,     // .................##..........##.
+    0x700e,     // .................###........###.
+    0x7ffe,     // .................##############.
+    0x3ffc,     // ..................############..
+    0x0,        // ................................
+  },
+//  {             // 245
+//     11, 22,
+//     0x00000000,  // ...........
+//     0x00000000,  // ...........
+//     0x00000000,  // ...........
+//     0x00000000,  // ...........
+//     0x000000C8,  // ...**..*...
+//     0x00000168,  // ..*.**.*...
+//     0x00000130,  // ..*..**....
+//     0x00000000,  // ...........
+//     0x000000F8,  // ...*****...
+//     0x000001FC,  // ..*******..
+//     0x0000018C,  // ..**...**..
+//     0x00000306,  // .**.....**.
+//     0x00000306,  // .**.....**.
+//     0x00000306,  // .**.....**.
+//     0x00000306,  // .**.....**.
+//     0x0000018C,  // ..**...**..
+//     0x000001FC,  // ..*******..
+//     0x000000F8,  // ...*****...
+//     0x00000000,  // ...........
+//     0x00000000,  // ...........
+//     0x00000000,  // ...........
+//     0x00000000,  // ...........
+//  },
   {             // 246
      11, 22,
      0x00000000,  // ...........
@@ -5425,80 +5500,155 @@
      0x00000000,  // ...........
   },
   {             // 249
-     10, 22,
-     0x00000000,  // ..........
-     0x00000000,  // ..........
-     0x00000000,  // ..........
-     0x00000000,  // ..........
-     0x00000060,  // ...**.....
-     0x00000030,  // ....**....
-     0x00000018,  // .....**...
-     0x00000000,  // ..........
-     0x00000186,  // .**....**.
-     0x00000186,  // .**....**.
-     0x00000186,  // .**....**.
-     0x00000186,  // .**....**.
-     0x00000186,  // .**....**.
-     0x00000186,  // .**....**.
-     0x00000186,  // .**....**.
-     0x0000018E,  // .**...***.
-     0x000001FE,  // .********.
-     0x000000F6,  // ..****.**.
-     0x00000000,  // ..........
-     0x00000000,  // ..........
-     0x00000000,  // ..........
-     0x00000000,  // ..........
-  },
+     16, 22,
+     0x0,    // ................
+     0xf800, // #####...........
+     0xcc00, // ##..##..........
+     0xfc00, // ######..........
+     0xf800, // #####...........
+     0xdc00, // ##.###..........
+     0x0,    // ................
+     0x1f80, // ...######.......
+     0x1f80, // ...######.......
+     0x1800, // ...##...........
+     0x1e00, // ...####.........
+     0x1800, // ...##...........
+     0x1f80, // ...######.......
+     0x1f80, // ...######.......
+     0x0,    // ................
+     0x3c0,  // ......####......
+     0x7e0,  // .....######.....
+     0xe60,  // ....###..##.....
+     0xc00,  // ....##..........
+     0xe60,  // ....###..##.....
+     0x7e0,  // .....######.....
+     0x3c0,  // ......####......
+  },
+//  {             // 249
+//     10, 22,
+//     0x00000000,  // ..........
+//     0x00000000,  // ..........
+//     0x00000000,  // ..........
+//     0x00000000,  // ..........
+//     0x00000060,  // ...**.....
+//     0x00000030,  // ....**....
+//     0x00000018,  // .....**...
+//     0x00000000,  // ..........
+//     0x00000186,  // .**....**.
+//     0x00000186,  // .**....**.
+//     0x00000186,  // .**....**.
+//     0x00000186,  // .**....**.
+//     0x00000186,  // .**....**.
+//     0x00000186,  // .**....**.
+//     0x00000186,  // .**....**.
+//     0x0000018E,  // .**...***.
+//     0x000001FE,  // .********.
+//     0x000000F6,  // ..****.**.
+//     0x00000000,  // ..........
+//     0x00000000,  // ..........
+//     0x00000000,  // ..........
+//     0x00000000,  // ..........
+//  },
   {             // 250
-     10, 22,
-     0x00000000,  // ..........
-     0x00000000,  // ..........
-     0x00000000,  // ..........
-     0x00000000,  // ..........
-     0x0000000C,  // ......**..
-     0x00000018,  // .....**...
-     0x00000030,  // ....**....
-     0x00000000,  // ..........
-     0x00000186,  // .**....**.
-     0x00000186,  // .**....**.
-     0x00000186,  // .**....**.
-     0x00000186,  // .**....**.
-     0x00000186,  // .**....**.
-     0x00000186,  // .**....**.
-     0x00000186,  // .**....**.
-     0x0000018E,  // .**...***.
-     0x000001FE,  // .********.
-     0x000000F6,  // ..****.**.
-     0x00000000,  // ..........
-     0x00000000,  // ..........
-     0x00000000,  // ..........
-     0x00000000,  // ..........
-  },
+        16, 22,
+        0x0,    // ................
+        0xc600, // ##...##.........
+        0xe600, // ###..##.........
+        0xf600, // ####.##.........
+        0xde00, // ##.####.........
+        0xce00, // ##..###.........
+        0xc600, // ##...##.........
+        0x0,    // ................
+        0x1fc0, // ...#######......
+        0x1fc0, // ...#######......
+        0x1800, // ...##...........
+        0x1f00, // ...#####........
+        0x1800, // ...##...........
+        0x1fc0, // ...#######......
+        0x1fc0, // ...#######......
+        0x0,    // ................
+        0x318,  // ......##...##...
+        0x318,  // ......##...##...
+        0x318,  // ......##...##...
+        0x318,  // ......##...##...
+        0x3f8,  // ......#######...
+        0x1f0,  // .......#####....
+  },
+//  {             // 250
+//     10, 22,
+//     0x00000000,  // ..........
+//     0x00000000,  // ..........
+//     0x00000000,  // ..........
+//     0x00000000,  // ..........
+//     0x0000000C,  // ......**..
+//     0x00000018,  // .....**...
+//     0x00000030,  // ....**....
+//     0x00000000,  // ..........
+//     0x00000186,  // .**....**.
+//     0x00000186,  // .**....**.
+//     0x00000186,  // .**....**.
+//     0x00000186,  // .**....**.
+//     0x00000186,  // .**....**.
+//     0x00000186,  // .**....**.
+//     0x00000186,  // .**....**.
+//     0x0000018E,  // .**...***.
+//     0x000001FE,  // .********.
+//     0x000000F6,  // ..****.**.
+//     0x00000000,  // ..........
+//     0x00000000,  // ..........
+//     0x00000000,  // ..........
+//     0x00000000,  // ..........
+//  },
   {             // 251
-     10, 22,
-     0x00000000,  // ..........
-     0x00000000,  // ..........
-     0x00000000,  // ..........
-     0x00000000,  // ..........
-     0x00000030,  // ....**....
-     0x00000078,  // ...****...
-     0x000000CC,  // ..**..**..
-     0x00000000,  // ..........
-     0x00000186,  // .**....**.
-     0x00000186,  // .**....**.
-     0x00000186,  // .**....**.
-     0x00000186,  // .**....**.
-     0x00000186,  // .**....**.
-     0x00000186,  // .**....**.
-     0x00000186,  // .**....**.
-     0x0000018E,  // .**...***.
-     0x000001FE,  // .********.
-     0x000000F6,  // ..****.**.
-     0x00000000,  // ..........
-     0x00000000,  // ..........
-     0x00000000,  // ..........
-     0x00000000,  // ..........
-  },
+        16, 22,
+        0x0,            // ................
+        0x78,           // ..........####..
+        0xfc,           // .........######.
+        0x1e6,          // ........####..##
+        0x1de,          // ........###.####
+        0x1e0,          // ........####....
+        0x70f8,         // ..###....#####..
+        0xf8c0,         // .#####...##.....
+        0x19dc0,        // ##..###.###.....
+        0x18f80,        // ##...#####......
+        0x1c780,        // ###...####......
+        0xcf00,         // .##..####.......
+        0x1f98,         // ....######..##..
+        0x39f0,         // ...###..#####...
+        0x70e0,         // ..###....###....
+        0xe000,         // .###............
+        0x1e000,        // ####............
+        0x1b000,        // ##.##...........
+        0x19800,        // ##..##..........
+        0x6600,         // ..##..##........
+        0x3b80,         // ...###.###......
+        0x0,            // ................
+  },
+//  {             // 251
+//     10, 22,
+//     0x00000000,  // ..........
+//     0x00000000,  // ..........
+//     0x00000000,  // ..........
+//     0x00000000,  // ..........
+//     0x00000030,  // ....**....
+//     0x00000078,  // ...****...
+//     0x000000CC,  // ..**..**..
+//     0x00000000,  // ..........
+//     0x00000186,  // .**....**.
+//     0x00000186,  // .**....**.
+//     0x00000186,  // .**....**.
+//     0x00000186,  // .**....**.
+//     0x00000186,  // .**....**.
+//     0x00000186,  // .**....**.
+//     0x00000186,  // .**....**.
+//     0x0000018E,  // .**...***.
+//     0x000001FE,  // .********.
+//     0x000000F6,  // ..****.**.
+//     0x00000000,  // ..........
+//     0x00000000,  // ..........
+//     0x00000000,  // ..........
+//     0x00000000,  // ..........
+//  },
   {             // 252
      10, 22,
      0x00000000,  // ..........
@@ -5525,78 +5675,165 @@
      0x00000000,  // ..........
   },
   {             // 253
-     10, 22,
-     0x00000000,  // ..........
-     0x00000000,  // ..........
-     0x00000000,  // ..........
-     0x00000000,  // ..........
-     0x0000000C,  // ......**..
-     0x00000018,  // .....**...
-     0x00000030,  // ....**....
-     0x00000000,  // ..........
-     0x00000186,  // .**....**.
-     0x00000186,  // .**....**.
-     0x00000186,  // .**....**.
-     0x000000CC,  // ..**..**..
-     0x000000CC,  // ..**..**..
-     0x000000CC,  // ..**..**..
-     0x00000048,  // ...*..*...
-     0x00000078,  // ...****...
-     0x00000030,  // ....**....
-     0x00000030,  // ....**....
-     0x00000030,  // ....**....
-     0x00000030,  // ....**....
-     0x000000E0,  // ..***.....
-     0x000000E0,  // ..***.....
-  },
+        16, 22,
+        0x0,    // ................
+        0x0,    // ................
+        0x0,    // ................
+        0x6006, // .##..........##.
+        0xf3cf, // ####..####..####
+        0xf7ef, // ####.######.####
+        0x3c3c, // ..####....####..
+        0x1998, // ...##..##..##...
+        0x318c, // ..##...##...##..
+        0x318c, // ..##...##...##..
+        0x30cc, // ..##....##..##..
+        0x38dc, // ..###...##.###..
+        0x1c38, // ...###....###...
+        0xff0,  // ....########....
+        0xff0,  // ....########....
+        0xc30,  // ....##....##....
+        0x1818, // ...##......##...
+        0x381c, // ..###......###..
+        0xfc3f, // ######....######
+        0xfc3f, // ######....######
+        0x0,    // ................
+        0x0,    // ................
+  },
+//  {             // 253
+//     10, 22,
+//     0x00000000,  // ..........
+//     0x00000000,  // ..........
+//     0x00000000,  // ..........
+//     0x00000000,  // ..........
+//     0x0000000C,  // ......**..
+//     0x00000018,  // .....**...
+//     0x00000030,  // ....**....
+//     0x00000000,  // ..........
+//     0x00000186,  // .**....**.
+//     0x00000186,  // .**....**.
+//     0x00000186,  // .**....**.
+//     0x000000CC,  // ..**..**..
+//     0x000000CC,  // ..**..**..
+//     0x000000CC,  // ..**..**..
+//     0x00000048,  // ...*..*...
+//     0x00000078,  // ...****...
+//     0x00000030,  // ....**....
+//     0x00000030,  // ....**....
+//     0x00000030,  // ....**....
+//     0x00000030,  // ....**....
+//     0x000000E0,  // ..***.....
+//     0x000000E0,  // ..***.....
+//  },
   {             // 254
-     11, 22,
-     0x00000000,  // ...........
-     0x00000000,  // ...........
-     0x00000000,  // ...........
-     0x00000000,  // ...........
-     0x00000300,  // .**........
-     0x00000300,  // .**........
-     0x00000300,  // .**........
-     0x00000300,  // .**........
-     0x00000378,  // .**.****...
-     0x000003FC,  // .********..
-     0x0000038C,  // .***...**..
-     0x00000306,  // .**.....**.
-     0x00000306,  // .**.....**.
-     0x00000306,  // .**.....**.
-     0x00000306,  // .**.....**.
-     0x0000038C,  // .***...**..
-     0x000003FC,  // .********..
-     0x00000378,  // .**.****...
-     0x00000300,  // .**........
-     0x00000300,  // .**........
-     0x00000300,  // .**........
-     0x00000300,  // .**........
-  },
-  {             // 255
-     10, 22,
-     0x00000000,  // ..........
-     0x00000000,  // ..........
-     0x00000000,  // ..........
-     0x00000000,  // ..........
-     0x000000CC,  // ..**..**..
-     0x000000CC,  // ..**..**..
-     0x00000000,  // ..........
-     0x00000000,  // ..........
-     0x00000186,  // .**....**.
-     0x00000186,  // .**....**.
-     0x00000186,  // .**....**.
-     0x000000CC,  // ..**..**..
-     0x000000CC,  // ..**..**..
-     0x000000CC,  // ..**..**..
-     0x00000048,  // ...*..*...
-     0x00000078,  // ...****...
-     0x00000030,  // ....**....
-     0x00000030,  // ....**....
-     0x00000030,  // ....**....
-     0x00000030,  // ....**....
-     0x000000E0,  // ..***.....
-     0x000000E0,  // ..***.....
-  },
+        16, 22,
+        0x0,    // ................
+        0x0,    // ................
+        0x0,    // ................
+        0x4,    // ..............#.
+        0x1180c,        // #...##.......##.
+        0x19c18,        // ##..###.....##..
+        0xd630, // .##.#.##...##...
+        0x7360, // ..###..##.##....
+        0x31c0, // ...##...###.....
+        0x19f0, // ....##..#####...
+        0x1f50, // ....#####.#.#...
+        0x1650, // ....#.##..#.#...
+        0x1f50, // ....#####.#.#...
+        0x19f0, // ....##..#####...
+        0x31c0, // ...##...###.....
+        0x7360, // ..###..##.##....
+        0xd630, // .##.#.##...##...
+        0x19c18,        // ##..###.....##..
+        0x1180c,        // #...##.......##.
+        0x4,    // ..............#.
+        0x0,    // ................
+        0x0,    // ................
+  },
+//  {             // 254
+//     11, 22,
+//     0x00000000,  // ...........
+//     0x00000000,  // ...........
+//     0x00000000,  // ...........
+//     0x00000000,  // ...........
+//     0x00000300,  // .**........
+//     0x00000300,  // .**........
+//     0x00000300,  // .**........
+//     0x00000300,  // .**........
+//     0x00000378,  // .**.****...
+//     0x000003FC,  // .********..
+//     0x0000038C,  // .***...**..
+//     0x00000306,  // .**.....**.
+//     0x00000306,  // .**.....**.
+//     0x00000306,  // .**.....**.
+//     0x00000306,  // .**.....**.
+//     0x0000038C,  // .***...**..
+//     0x000003FC,  // .********..
+//     0x00000378,  // .**.****...
+//     0x00000300,  // .**........
+//     0x00000300,  // .**........
+//     0x00000300,  // .**........
+//     0x00000300,  // .**........
+//  },
+   {             // 255
+     16, 22,
+     0x0,            // ................
+     0x7f80,         // ..########......
+     0xffc0,         // .##########.....
+     0x1ffe0,        // ############....
+     0x1f3e0,        // #####..#####....
+     0x1e1e0,        // ####....####....
+     0x1c0e0,        // ###......###....
+     0x1c0e0,        // ###......###....
+     0x1c0e0,        // ###......###....
+     0x1c0e0,        // ###......###....
+     0x1c0e0,        // ###......###....
+     0x1c0e0,        // ###......###....
+     0x1c0e0,        // ###......###....
+     0x1c0e0,        // ###......###....
+     0x1c0e0,        // ###......###....
+     0x1c7fc,        // ###...#########.
+     0x1c3f8,        // ###....#######..
+     0x1c1f0,        // ###.....#####...
+     0x1c0e0,        // ###......###....
+     0x40,           // ..........#.....
+     0x0,            // ................
+     0x0,            // ................
+  },
+//  {             // 255
+//     10, 22,
+//     0x00000000,  // ..........
+//     0x00000000,  // ..........
+//     0x00000000,  // ..........
+//     0x00000000,  // ..........
+//     0x000000CC,  // ..**..**..
+//     0x000000CC,  // ..**..**..
+//     0x00000000,  // ..........
+//     0x00000000,  // ..........
+//     0x00000186,  // .**....**.
+//     0x00000186,  // .**....**.
+//     0x00000186,  // .**....**.
+//     0x000000CC,  // ..**..**..
+//     0x000000CC,  // ..**..**..
+//     0x000000CC,  // ..**..**..
+//     0x00000048,  // ...*..*...
+//     0x00000078,  // ...****...
+//     0x00000030,  // ....**....
+//     0x00000030,  // ....**....
+//     0x00000030,  // ....**....
+//     0x00000030,  // ....**....
+//     0x000000E0,  // ..***.....
+//     0x000000E0,  // ..***.....
+//  },
   };
+
+// WarEagle-Symbol-Patch
+// Zeichen     Symbol
+//
+//     255     Pfeil
+//     254     Lautsprecher
+//     253     Wecker
+//     251     Maennchen
+//     250     "NEU"
+//     249     "REC"
+//     244     Weiter-Links
+//     245     TV
diff -Naur vdr-1.3.22/i18n.c vdr-1.3.22-icon/i18n.c
--- vdr-1.3.22/i18n.c	2005-02-27 10:45:57.000000000 +0100
+++ vdr-1.3.22-icon/i18n.c	2005-03-06 02:04:44.000000000 +0100
@@ -2611,6 +2611,35 @@
     "Teema",
     "Tema",
   },
+  { "Setup.OSD$WarEagle icons",
+    "WarEagle-Symbole",
+    "", // Slovenski                                                           
+    "", // Italiano                                                            
+    "", // Nederlands                                                          
+    "", // Português                                                           
+    "", // Français                                                            
+    "", // Norsk                                                               
+    "", // Suomi                                                               
+    "", // Polski                                                              
+    "", // Español                                                             
+    "", // Greek                                                               
+    "", // Svenska                                                             
+    "", // Romaneste                                                           
+    "", // Magyar                                                              
+    "", // Català                                                              
+#if VDRVERSNUM > 10302                                                         
+    "", // Russian                                                             
+#if VDRVERSNUM > 10307                                                         
+    "", // Hrvatski                                                            
+#if VDRVERSNUM > 10313                                                         
+    "", // Eesti                                                               
+#if VDRVERSNUM > 10316                                                         
+    "", // Dansk                                                               
+#endif                                                                         
+#endif                                                                         
+#endif                                                                         
+#endif                                                                         
+  },                                                                           
   { "Setup.OSD$Left",
     "Links",
     "",// TODO
diff -Naur vdr-1.3.22/iconpatch.h vdr-1.3.22-icon/iconpatch.h
--- vdr-1.3.22/iconpatch.h	1970-01-01 01:00:00.000000000 +0100
+++ vdr-1.3.22-icon/iconpatch.h	2005-03-04 19:58:28.000000000 +0100
@@ -0,0 +1,24 @@
+/*
+ * iconpatch.h: Information of iconpatch
+ *
+ * Diese Datei ist die Übersichtsdatei für den Iconpatch.
+ * Hier werden kleine Infos abgelegt.
+ * Der Iconpatch ändert die Dateien:
+ *   iconpatch.h
+ *   menu.c
+ *   recording.c
+ *   fontosd.c
+ *
+ */
+
+//Iconpatch-Variablen - Anfang
+#define ICON_RADIO         char(244)
+#define ICON_TV            char(245)
+#define ICON_TV_VERSCHL    char(239)
+#define ICON_REC           char(249)
+#define ICON_NEU           char(250)
+#define ICON_RENNER        char(251)
+#define ICON_UHR           char(253)
+#define ICON_LAUTSPRECHER  char(254)
+#define ICON_PFEIL         char(255)
+//Iconpatch-Variablen - Ende
diff -Naur vdr-1.3.22/menu.c vdr-1.3.22-icon/menu.c
--- vdr-1.3.22/menu.c	2005-02-27 15:09:00.000000000 +0100
+++ vdr-1.3.22-icon/menu.c	2005-03-07 03:57:22.000000000 +0100
@@ -8,6 +8,7 @@
  */
 
 #include "menu.h"
+#include "iconpatch.h"
 #include <ctype.h>
 #include <limits.h>
 #include <stdio.h>
@@ -371,6 +372,14 @@
        } else {
         if (sortMode == csmProvider || sortMode == csmProviderNumber)
            asprintf(&buffer, "%d\t%s - %s", channel->Number(), channel->Provider(), channel->Name());
+        else if (Setup.WarEagleIcons) {
+           if (channel->Vpid() == 1 || channel->Vpid() == 0)
+              asprintf(&buffer, "%d\t%c %-30s", channel->Number(), ICON_RADIO, channel->Name());
+           else if (channel->Ca() == 0)
+              asprintf(&buffer, "%d\t%c %-30s", channel->Number(), ICON_TV, channel->Name());
+           else
+              asprintf(&buffer, "%d\t%c %-30s", channel->Number(), ICON_TV_VERSCHL, channel->Name());
+        }
         else
            asprintf(&buffer, "%d\t%s", channel->Number(), channel->Name());
        }
@@ -735,7 +744,7 @@
 {
   char *buffer = NULL;
   asprintf(&buffer, "%c\t%d\t%s%s%s\t%02d:%02d\t%02d:%02d\t%s",
-                    !(timer->HasFlags(tfActive)) ? ' ' : timer->FirstDay() ? '!' : timer->Recording() ? '#' : '>',
+                    !(timer->HasFlags(tfActive)) ? ' ' : timer->FirstDay() ? Setup.WarEagleIcons ? ICON_PFEIL : '!' : timer->Recording() ? Setup.WarEagleIcons ? ICON_REC : '#' : Setup.WarEagleIcons ? ICON_UHR : '>',
                     timer->Channel()->Number(),
                     timer->IsSingleEvent() ? *WeekDayName(timer->StartTime()) : "",
                     timer->IsSingleEvent() ? " " : "",
@@ -1762,6 +1771,7 @@
   Add(new cMenuEditStraItem(tr("Setup.OSD$Skin"),                   &skinIndex, numSkins, skinDescriptions));
   if (themes.NumThemes())
   Add(new cMenuEditStraItem(tr("Setup.OSD$Theme"),                  &themeIndex, themes.NumThemes(), themes.Descriptions()));
+  Add(new cMenuEditBoolItem(tr("Setup.OSD$WarEagle icons"),         &data.WarEagleIcons));
   Add(new cMenuEditIntItem( tr("Setup.OSD$Left"),                   &data.OSDLeft, 0, MAXOSDWIDTH));
   Add(new cMenuEditIntItem( tr("Setup.OSD$Top"),                    &data.OSDTop, 0, MAXOSDHEIGHT));
   Add(new cMenuEditIntItem( tr("Setup.OSD$Width"),                  &data.OSDWidth, MINOSDWIDTH, MAXOSDWIDTH));
diff -Naur vdr-1.3.22/recording.c vdr-1.3.22-icon/recording.c
--- vdr-1.3.22/recording.c	2005-02-12 11:17:47.000000000 +0100
+++ vdr-1.3.22-icon/recording.c	2005-03-04 20:03:31.000000000 +0100
@@ -8,6 +8,7 @@
  */
 
 #include "recording.h"
+#include "iconpatch.h"
 #include <dirent.h>
 #include <errno.h>
 #include <fcntl.h>
@@ -492,7 +493,7 @@
 
 const char *cRecording::Title(char Delimiter, bool NewIndicator, int Level) const
 {
-  char New = NewIndicator && IsNew() ? '*' : ' ';
+  char New = NewIndicator && IsNew() ? Setup.WarEagleIcons ? ICON_NEU : '*' : ' ';
   free(titleBuffer);
   titleBuffer = NULL;
   if (Level < 0 || Level == HierarchyLevels()) {
